test/unit/user_test.rb

```diff
@@ -93,6 +93,91 @@ class UserTest < ActiveSupport::TestCase
     end
   end
 
+  test 'send_daily_triage! sends email after preferred email_time_of_day' do
+    user  = users(:morning_person)
+    repo  = repos(:rails_rails)
+
+    issue = repo.issues.new
+    issue.title           = "Foo Bar"
+    issue.url             = "http://schneems.com"
+    issue.last_touched_at = 2.days.ago
+    issue.state           = 'open'
+    issue.html_url        = "http://schneems.com"
+    issue.number          = (repo.issues.last.try(:number) || 0) + 1
+    issue.save
+
+    repo_sub      = user.repo_subscriptions.new
+    repo_sub.repo = repo
+    repo_sub.save
+
+    now = Time.now
+    Timecop.freeze Time.new(now.year, now.month, now.day, 06, 0, 0) do
+      assert_difference("User.find(#{user.id}).issue_assignments.where(delivered: true).count", 1) do
+        Issue.any_instance.stubs(:valid_for_user?).returns(true)
+        user.send_daily_triage!
+      end
+    end
+  end
+
+  test 'send_daily_triage! does not send email before preferred email_time_of_day' do
+    user  = users(:morning_person)
+    repo  = repos(:rails_rails)
+
+    issue = repo.issues.new
+    issue.title           = "Foo Bar"
+    issue.url             = "http://schneems.com"
+    issue.last_touched_at = 2.days.ago
+    issue.state           = 'open'
+    issue.html_url        = "http://schneems.com"
+    issue.number          = (repo.issues.last.try(:number) || 0) + 1
+    issue.save
+
+    repo_sub      = user.repo_subscriptions.new
+    repo_sub.repo = repo
+    repo_sub.save
+
+    now = Time.now
+    Timecop.freeze Time.new(now.year, now.month, now.day, 02, 0, 0) do
+      assert_difference("User.find(#{user.id}).issue_assignments.where(delivered: true).count", 0) do
+        Issue.any_instance.stubs(:valid_for_user?).returns(true)
+        user.send_daily_triage!
+      end
+    end
+  end
+
+  test 'send_daily_triage! can be scheduled to run multiple times a day' do
```

Comment:
Thanks @MonikaMahanthappa. The tests above you refer too only run the send_daily_triage! method once.

This test was to ensure that if we call send_daily_triage! multiple times after the preferred time we won't send multiple emails. Since this is the responsibility of EmailDecider I agree it is a redundant.